api/v1alpha1/conversion_utils.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ConversionAnnotationKey is the annotation key to ensure lossless
+// conversion of objects between API versions.
+const ConversionAnnotationKey = "vmoperator.vmware.com/conversion"
```

Comment:
Since this is just an annotation, it can be changed by users (accidentally, or deliberately).  I am curious what the upstream pattern on this is?  Should we have some kind of webhook prevent users from setting this annotation?

Reply:
[Upstream kubernetes](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) has certain reserved prefixes such as `kubernetes.io/` and `k8s.io` which are reserved for use by core components.
CAPI has a similar prefix `node.cluster.x-k8s.io` which is owned and operated by their components. 

Following convention and similar usages in the codebase, used the annotation prefixed with `vmoperator.vmware.com` to declare the intent that it is used internally by VMOp controllers.

Reply:
As for the suggestion of preventing users from setting these annotations, webhooks will intercept all requests and would not differentiate between user generated request v/s conversion function update requests.